### DISCUSSION TITLE
Is there any way to get the height of a panel programmatically?

### DISCUSSION THREAD

**Discussion by ZeroIntensity** [Textualize/rich#1767] [Q&A] [answered]
I need to get the height of a panel inside of a layout. I've looked through the documentation and wasn't able to find anything.
`Panel.height` exists, but that returns `None` if it wasn't set manually.

Here's an example of what I am trying to do:

```py
from rich.layout import Layout
from rich.panel import Panel

layout = Layout()

layout.split_column(
    Panel("something"),
    Panel("another_thing")
)

something_height: int = layout["something"].height
```

**Comment by willmcgugan** [comment, marked as answer]
The `Panel.height` is used to force the height in advance. It's not to report the height as it is rendered.

If you want to know the available height within a Panel you can define a console renderable with a `__rich_console__` method. See the [docs](https://rich.readthedocs.io/en/latest/protocol.html) for how to define a console renderable. The `options` parameter will have a `max_height` attribute which is the number of lines available.
